### PR TITLE
Update rust toolchain to 1.68.2

### PR DESCRIPTION
**Pull request by roypat**
## Changes

- Update rust toolchain
- Appease new clippy lints wrt derivable `Default` implementations and elidable lifetimes
- Add `sched_yield` to the seccomp allow list, as with 1.67.0, the rust standard library changed the backend for `mpmc` channels (which we use through `std::sync::mpsc` for inter-thread communication) to use crossbeams implementatation, which uses `sched_yield` as part of its spin-locking logic [[1]]

[1]: https://github.com/rust-lang/rust/blame/cf32b9de1e8f66526c36ad2927458558d2e81093/library/std/src/sync/mpmc/utils.rs#L130

## Reason

backport of #3576

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [x] If a specific issue led to this PR, this PR closes the issue.
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] API changes follow the [Runbook for Firecracker API changes][2].
- [x] User-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.
- [x] New `TODO`s link to an issue.
- [x] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [ ] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: src/mmds/src/data_store.rs [hunk 1]
```diff
@@ -19,18 +19,13 @@ pub struct Mmds {
 }
 
 /// MMDS version.
-#[derive(Clone, Copy, Debug, PartialEq, Eq, Deserialize, Serialize)]
+#[derive(Clone, Copy, Debug, Default, PartialEq, Eq, Deserialize, Serialize)]
 pub enum MmdsVersion {
+    #[default]
```
Comment (pb8o) [inline src/mmds/src/data_store.rs:24]: out of scope, but shouldn't the default be V2?

**Comment by JonathanWoollett-Light** [comment]
All the changes from these commits are required to pass CI, as a result of the rust update, I think it would make sense in this case for these to all be in 1 commit, so we minimize commits which don't pass CI.

File: resources/seccomp/aarch64-unknown-linux-musl.json
Comment (pb8o) [inline resources/seccomp/aarch64-unknown-linux-musl.json:1]: Did we check it is used by all the threads?

File: src/mmds/src/data_store.rs (hunk 1 above)
Comment (roypat) [reply src/mmds/src/data_store.rs:24]: Yeah, this surprised me too, but the old manual `Default` implementation returned `V1`. The `default` impl seemingly is only used in test anyway though

File: resources/seccomp/aarch64-unknown-linux-musl.json
Comment (roypat) [reply resources/seccomp/aarch64-unknown-linux-musl.json:1]: Yeah, all threads use channels to communicate. Api <-> VMM to send instructions to VMM thread, and VMM <-> Vcpu for things such as pausing, resuming, etc.

Comment (pb8o) [reply resources/seccomp/aarch64-unknown-linux-musl.json:1]: Thanks!

**Comment by bchalios** [comment]
> All the changes from these commits are required to pass CI, as a result of the rust update, I think it would make sense in this case for these to all be in 1 commit, so we minimize commits which don't pass CI.

I don't think we want to go that way. We currently do not ensure that all commits pass CI (see code coverage changes, binary size changes, etc.). Also, I see value in having these changes in different commits. It is much easier to review and I identify issues later on.

**Comment by roypat** [comment]
> > All the changes from these commits are required to pass CI, as a result of the rust update, I think it would make sense in this case for these to all be in 1 commit, so we minimize commits which don't pass CI.
> 
> I don't think we want to go that way. We currently do not ensure that all commits pass CI (see code coverage changes, binary size changes, etc.). Also, I see value in having these changes in different commits. It is much easier to review and identify issues later on.

ah whoops, missed this earlier. I agree with Babis here. Having it broken into logical pieces makes cherry-picking/backporting a lot easier. For example, the coverage adjustment is different for every release branch, but the seccomp changes applied cleanly to all of them. With clippy, there was no knowing if I could cherry-pick from here (which I could), or whether old branches would need different patches.

**Review by JonathanWoollett-Light — APPROVED** [review]

**Pull request merged**